### PR TITLE
[Compaction] track RowsetReader's mem & add metric

### PR DESCRIPTION
**Pull request by vagetablechicken**
Ref https://github.com/apache/incubator-doris/issues/3624#issuecomment-655933244
Only RowsetReaders in compaction are under the track.
Other RowsetReaders won't be effected, because the parent_tracker is nullptr.

### CODE REVIEW

File: be/src/olap/compaction.h [hunk 1]
```diff
@@ -77,6 +77,8 @@ class Compaction {
     int64_t _get_input_num_rows_from_seg_grps();
 
 protected:
+    MemTracker _mem_tracker;
```
[the remark this answers is not in the document]
Comment (vagetablechicken) [reply be/src/olap/compaction.h:81]: done

**Review by morningman — COMMENTED** [review]
Hi @vagetablechicken , Plz add this new metrics to the document `be-metrics.md`, Thank you

**Comment by vagetablechicken** [comment]
> Hi @vagetablechicken , Plz add this new metrics to the document `be-metrics.md`, Thank you

Finished.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**